packages/tocco-ui/src/Panel/Panel.js

```diff
@@ -14,12 +14,21 @@ class Panel extends React.PureComponent {
 
   toggleOpenState = () => {
     if (this.props.isToggleable) {
+      if (typeof this.props.onToggle === 'function') {
+        this.props.onToggle(!this.state.isOpen)
+      }
       this.setState(prevState => ({
         isOpen: !prevState.isOpen
       }))
     }
   }
 
+  componentDidUpdate() {
+    if (typeof this.props.onToggle === 'function') {
+      this.setState({isOpen: this.props.isOpen})
+    }
```

Comment:
- the local state should be updated with the `isOpen` prop, no matter if there's an `onToggle` callback or not
- I'm not sure if `componentDidUpdate` is the right lifecycle method to use. Reading the documentation, I think `getDerivedStateFromProps` could be the better choice (https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops).
- If using `componentDidUpdate` is the way to go, I think `setState` should only be called if the prop is different from the current state (for performance reasons)

Reply:
getDerivedStateFromProps seems to be the better choice because getDerivedStateFromProps has no sideeffects (see http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/) and is executed before render. Hence Panel is rendered one time less. Also the constructor is not needed because getDerivedStateFromProps is executed while mounting and updating.

Reply:
OK, looks better now.

One more thing: IMO, a function that is supposed to return something (whether an actual value, `null` or `undefined`), should return a return value explicitly in all cases. Here, if the condition on line 23 doesn't evaluate to true, `undefined` is returned implicitly (nobody knows whether that's intended or you just forgot the return value). So, I suggest to add `return null` on line 26 so that it's clear immediately.